### PR TITLE
DEPR: Deprecate DataFrame.last and Series.last

### LINKED ISSUE #53692: DEPR: deprecate DataFrame.last and Series.last
just opening this so it's not forgotten - xref https://github.com/pandas-dev/pandas/pull/53419

### PR DESCRIPTION
**Pull request by rmhowe425**
- [X] closes #53692 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [X] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by rmhowe425** [comment]
@MarcoGorelli Is something going on with the CI environment? The tests that are failing for this PR are not failing locally for me when I run pytest. 

The same tests also started failing in #53439 this morning after I updated my entry in whatsnew v2.1.0

File: pandas/core/generic.py [hunk 1]
```diff
@@ -228,6 +228,7 @@ class NDFrame(PandasObject, indexing.IndexingMixin):
     N-dimensional analogue of DataFrame. Store multi-dimensional in a
     size-mutable, labeled data structure
 
+    Parameters
     Parameters
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:232]: no need to duplicate this

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for your pr, generally looks good

failures might be unrelated - just got a minor comment, if you address that and then fetch and merge upstream/main then hopefully it'll be resolved

**Comment by rmhowe425** [comment]
@MarcoGorelli The Doc bulid test failure seems to be unrelated

The docstring test is failing because the warning message that I imeplemented is executing. Since this method is being deprecated what would be the recommended solution in this instance? 

Should I add an additional note in the docstring to  create a mask and filter using `.loc` instead of `last` and then add code that shows how to do that? I don't believe this was handled in the PR for deprecating [series.first and DataFrame.first](https://github.com/pandas-dev/pandas/pull/53419/files#diff-1a2e3df0db7dd8bddc2ec4bff9de8a7a55e328e6c32e2cecde761dc9549fcd46)

**Comment by MarcoGorelli** [comment]
thanks @rmhowe425 - does it work to `doctest: skip` the failing doctest?

**Review by MarcoGorelli — APPROVED** [review]
nice one @rmhowe425 !

**Pull request merged**